src/app/plugins/controller.plugin.ts

```diff
@@ -0,0 +1,27 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, ChannelType, IMessage } from '../../common/types';
+
+export class PluginControl extends Plugin {
+  public name: string = 'controller';
+  public description: string = 'Controls activating and deactivating plugins.';
+  public usage: string = 'controller <activate | deactive> <plugin name>';
+  public permission: ChannelType = ChannelType.Admin;
+  public commandPattern: RegExp = /^(deactivate|activate) (?!\s*$).+/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]): Promise<void> {
+    const [method, pluginName] = args;
+
+    try {
+      await this.container.pluginService.setPluginState(this.container, pluginName, method === 'activate') ;
+    } catch(e) {
+      await message.channel.send(e.message);
+      return;
+    }
```

Comment:
instead of try catch, just do `this.container.pluginService.setPluginState(this.container, pluginName, method === 'activate').catch((e) => ...`

Reply:
The issue here is that I want to return from the function after I catch an error, this isn't possible with the `.catch()` syntax.